src/common/hooks/useGeolocatedRegions.ts

```diff
@@ -0,0 +1,27 @@
+import { getGeolocatedRegions } from 'common/regions';
+import useGeolocation from './useGeolocation';
+import useCountyToZipMap from './useCountyToZipMap';
+
+/**
+ * Hook to provide a map of region type (metroArea, county, state) to region
+ * based on the user's geolocation information.
+ * `isLoading` will be true until the data is available.
+ * `userRegions` will be null until loaded or if unavailable after loading
+ */
+const useGeolocatedRegions = () => {
```

Comment:
This allows us to avoid calling `useGeolocation()` and `useCountyToZipMap()` in many places, which reduces the places we'll need to change to load just the necessary zip data.